### ISSUE TITLE
`enforce-consistent-variable-syntax` fixes incorrectly with `--spacing()`

### ISSUE THREAD

**Issue by kimkanu** [schoero/eslint-plugin-readable-tailwind#305]
### Description

In some cases, `enforce-consistent-variable-syntax` seems to fix some arbitrary value including `--spacing()` incorrectly.

### Flavor

TSX

### Code Input

```tsx
grid-cols-[--spacing(24)_1fr]
```

### Expected Behavior

```tsx
No fixes
```

### Actual Behavior

```tsx
It got changed into `grid-cols-(--spacing(24)_1fr)`, which is invalid:

.grid-cols-\(--spacing\(24\)_1fr\) {
  grid-template-columns: var(calc(var(--spacing) * 24) 1fr);
}
```

### ESLint Log

```shell
warning  Incorrect variable syntax: grid-cols-[--spacing(24)_1fr]  better-tailwindcss/enforce-consistent-variable-syntax
```

### ESLint Config

```typescript
{
    settings: {
      "better-tailwindcss": {
        entryPoint: resolve(
          dirname(fileURLToPath(import.meta.url)),
          "../ui/src/styles/globals.css",
        ),
      },
    },
    plugins: {
      "better-tailwindcss": pluginBetterTailwindcss,
    },
    rules: {
      ...pluginBetterTailwindcss.configs["stylistic-warn"].rules,
      ...pluginBetterTailwindcss.configs["correctness-error"].rules,
      "better-tailwindcss/enforce-canonical-classes": "error",
      "better-tailwindcss/no-unknown-classes": [
        "warn",
        { ignore: ["toaster", "group"] },
      ],
      "better-tailwindcss/enforce-consistent-line-wrapping": [
        "warn",
        { strictness: "loose" },
      ],
    },
  },
```

### Versions

- ESLint: 9.35.0
- Parser: `@typescript-eslint/parser` 8.43.0
- Plugin: 4.0.1
- Node: v22.17.0
